### DISCUSSION TITLE
custom search command

### DISCUSSION THREAD

**Discussion by Isos9** [neovim/neovim#29264] [General]
Hi,

I was wondering if there was a way to implement a custom search command like `:%s/match/replace/` but linked to another program that will search, that could look like `:%sg/match/replace/` or maybe just an override of the current search builtin that will just use another tool instead of the built-in one.

Please let me know, thanks

**Comment by yochem** [comment]
You can process a buffer using any command line program. For example with sed:

```
:.!sed 's/pattern/replace' %
```
